test/gtest/Makefile.am

```diff
@@ -134,13 +135,18 @@ gtest_LDFLAGS  += $(HIP_LDFLAGS)
 gtest_LDADD    += $(HIP_LIBS)
 endif
 
-
-noinst_HEADERS =            \
-	common/gtest.h          \
-	common/test.h           \
-	common/test_ucc.h       \
-	core/test_context.h     \
-	core/test_mc_reduce.h   \
+gtest_CXXFLAGS += $(UCX_CXXFLAGS)
+gtest_CPPFLAGS += $(UCX_CPPFLAGS)
+gtest_LDFLAGS  += $(UCX_LDFLAGS)
+gtest_LDADD    += $(UCX_LIBS) $(UCX_LIBADD)
```

Comment:
UCC is by prupose independent from UCX, I think we should not link UCX libs directly here. CC @manjugv

Reply:
I linked UCX here in order to use ucp functions in the gtest. It's required that exported memory handles are passed to the sliding window algorithm, so to export them I need to call ucp_mem_map.

Reply:
Alternatively, we'll have to build a 3rd party app specificaly for this test? Either way, if we can't link against UCX then we can't test this. We need a way to export memhandles. What do you guys suggest we do?

Reply:
Adding conditional compilation should be ok

Reply:
I wrapped this snippet with HAVE_UCX, which only gets defined if UCC is compiled with UCX.